File: Siren.java
import javafx.scene.media.Media;
import javafx.scene.media.MediaPlayer;
import javafx.scene.shape.Polygon;

public class Siren {
    public Siren (String mediaURL){
        Media media = new Media(mediaURL);
        mediaPlayer = new MediaPlayer(media);
        mediaPlayer.setCycleCount(MediaPlayer.INDEFINITE);
        view = new SirenView();
    }
    public void play(){
        view.startBlinking();
        mediaPlayer.play();
    }
    public void stop(){
        view.stopBlinking();
        mediaPlayer.stop();
    }
    public Polygon getView() {
        return view;
    }
    private SirenView view;
    private MediaPlayer mediaPlayer;
}

File: siren.cpp
#include "siren.h"
#include "sirenview.h"
#include <iostream>

Siren::Siren(){

}
Siren::Siren(SirenView * view): view(view) {
    isActive = false;
}
void Siren::play() {
    isActive = true;
    view->startBlinking();
}
void Siren::stop() {
    isActive = false;
    view->stopBlinking();
}
